devops/jobs/ImageBuilder.groovy

```diff
@@ -26,54 +26,61 @@ import static org.edx.jenkins.dsl.Constants.common_wrappers
 
 class ImageBuilder {
     public static def job = { dslFactory, extraVars ->
-        dslFactory.job(extraVars.get("FOLDER_NAME", "DockerCI") + "/image-builder") {
+        //assertions come earlier to ensure APPS_TO_CONFIG can be read
+        assert extraVars.containsKey('APPS_TO_CONFIG'): "Please define APPS_TO_CONFIG"
+        assert extraVars.containsKey('CONFIGURATION_REPO_URL') : "Please define CONFIGURATION_REPO_URL"
 
-            assert extraVars.containsKey('CONFIGURATION_REPO_URL') : "Please define CONFIGURATION_REPO_URL"
+        // for each application
+        extraVars.get("APPS_TO_CONFIG").each { app_name, app_config ->
+            dslFactory.job(extraVars.get("FOLDER_NAME", "DockerCI") + "/image-builders/" + app_name + "-image-builder") {
 
-            def config_branch = 'master'
+                def config_branch = 'master'
 
-            wrappers common_wrappers
-            // add credentials to log in to DockerHub; ID refers to credential ID as set up in Jenkins
-            wrappers {
-                credentialsBinding {
-                    file("CONFIG_JSON_FILE", extraVars.get("CONFIG_JSON_FILE_CREDENTIAL_ID", "docker-config-json"))
+                wrappers common_wrappers
+                // add credentials to log in to DockerHub; ID refers to credential ID as set up in Jenkins
+                wrappers {
+                    credentialsBinding {
+                        file("CONFIG_JSON_FILE", extraVars.get("CONFIG_JSON_FILE_CREDENTIAL_ID", "docker-config-json"))
+                    }
+                    // add master commits of application repository and configuration repository that are used to build the image
+                    buildName('#${BUILD_NUMBER} with app. version = ${ENV:0:7, var="APP_VERSION"} and config. version = ${GIT_REVISION, length=8}')
```

Comment:
Also, I know this might not be as useful since each application will have their own builder job now, but I thought it might still be helpful to know which commits were checked out when building the image.